components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -267,15 +270,28 @@ class SitePermissionsFeature(
         permissionRequest: PermissionRequest,
         session: Session
     ): SitePermissionsDialogFragment? {
-        val action = requireNotNull(sitePermissionsRules).getActionFrom(permissionRequest)
-        return when (action) {
-            SitePermissionsRules.Action.BLOCKED -> {
-                permissionRequest.reject()
-                session.contentPermissionRequest.consume { true }
-                null
-            }
-            SitePermissionsRules.Action.ASK_TO_ALLOW -> {
-                createPrompt(permissionRequest, session)
+        // For now we only support autoplay via sitePermissionsRules until we add support for
+        // autoplay for specific sites see Fenix issue  #8603
+        return if (permissionRequest.isForAutoplay() && sitePermissionsRules == null) {
+            permissionRequest.reject()
+            session.contentPermissionRequest.consume { true }
+            null
+        } else {
```

Comment:
After we add support for adding autoplay settings per site, we can delete this part :)